src/main/java/com/google/sps/data/Journey.java

```diff
@@ -0,0 +1,180 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Map;
+import java.lang.reflect.Array;
+import com.google.maps.errors.ApiException;
+import java.io.IOException;
+import java.lang.InterruptedException;
+
+import java.util.HashMap;
+
+/**
+ * Class that represents a delivery journey.
+ */
+public class Journey {
+  private int minTime;
+  private int numberOfWaypoints;
+  private ArrayList<Point> orderedWaypoints;
+  // Variable pathFinder will be used for getting the path between two Point objects.
+  // private PathFinder pathFinder;
+  private ArrayList<Pair> restrictions;
+  private Point start;
```

Comment:
The start point is not added in the `waypoints `hashmap nor in `orderedWaypoints` because his position is fixed and the algorithm I implemented will try all valid orders of visiting waypoints.
I will change it to be the last element in `orderedWaypoints`. It is easier to get the order starting from the last point to be visited, but I can reverse the array if it is bad practice/design.